src/contract.jl

```diff
@@ -73,7 +73,11 @@ function logscalar(
     denominator_terms
   end
 
-  return sum(log.(numerator_terms)) - sum(log.((denominator_terms)))
+  if any(t -> iszero(t), collect(denominator_terms))
+    return -Inf
+  else
+    return sum(log.(numerator_terms)) - sum(log.((denominator_terms)))
+  end
```

Comment:
```suggestion
  any(iszero, denominator_terms) && return -Inf
  return sum(log.(numerator_terms)) - sum(log.((denominator_terms)))
```
Not sure if `collect(denominator_terms)` can be changed to `denominator_terms`, what's the type of `denominator_terms`?

Reply:
It's a map from `partitionedges` to scalars so yeah I think the `collect()` can be dropped. Have changed that.

Reply:
You removed `collect` but didn't simplify the code in the other ways that I suggested.

Reply:
Ah I see sorry I missed that.